src/Microsoft.PowerShell.Commands.Management/commands/management/ClearRecycleBinCommand.cs

```diff
@@ -224,25 +226,14 @@ private void EmptyRecycleBin(string drivePath)
                 progress.RecordType = ProgressRecordType.Processing;
                 WriteProgress(progress);
 
+                // no need to check result as a failure is returned only if recycle bin is already empty
                 uint result = NativeMethod.SHEmptyRecycleBin(IntPtr.Zero, drivePath,
                                                             NativeMethod.RecycleFlags.SHERB_NOCONFIRMATION |
                                                             NativeMethod.RecycleFlags.SHERB_NOPROGRESSUI |
                                                             NativeMethod.RecycleFlags.SHERB_NOSOUND);
-                int lastError = Marshal.GetLastWin32Error();
-
-                // update the progress bar to completed
                 progress.PercentComplete = 100;
                 progress.RecordType = ProgressRecordType.Completed;
                 WriteProgress(progress);
-
-                // 0 is for a successful operation
-                // 203 comes up when trying to empty an already emptied recyclebin
-                // 18 comes up when there are no more files in the given recyclebin
-                if (!(lastError == 0 || lastError == 203 || lastError == 18))
```

Comment:
Was it inter-process conflict fix?

Reply:
Looking at the docs, this API doesn't rely on GetLastError().  This existing code seems to have been mistaken.  The API only returns an HRESULT.